js/pathfinding.js

```diff
@@ -5,10 +5,13 @@ import {settingsKey} from "./settings.js";
 import {getSnapPointForTokenObj, iterPairs} from "./util.js";
 
 import * as GridlessPathfinding from "../wasm/gridless_pathfinding.js";
-import {PriorityQueueSet} from "./data_structures.js";
+import {MaxSizeStackSet, PriorityQueueSet} from "./data_structures.js";
 
-let cachedNodes = undefined;
-let cacheElevation;
+const maxCacheIds = 5;
+const cache = {
+	nodes: new Map(),
+	cacheIds: new MaxSizeStackSet(maxCacheIds)
```

Comment:
Maybe instead of a general `MaxSizeStackSet` that won't be used anywhere else anyway, we could have a `Cache` class that handles all of this stuff internally. It could also then do the deletion of old entries atomically instead of relying on extra code by the caller.